### PR TITLE
Add app icon to MacOSExample

### PR DESCRIPTION
**Pull request by jfedak**
## Summary

This PR adds an app icon to `MacOSExample` app. The icon was created according to [Apple interface guidelines](https://developer.apple.com/design/human-interface-guidelines/app-icons), using the provided [template](https://developer.apple.com/design/resources/#macos-apps).

## Test plan

The icon displays properly:

- in Dock
DockExample

- in Stage Manager
StageManagerExample

- in Spotlight
SpotlightExample

### CODE REVIEW

File: MacOSExample/macos/MacOSExample-macOS/Assets.xcassets/AppIcon.appiconset/256 1.png
Comment (tomekzaw) [inline MacOSExample/macos/MacOSExample-macOS/Assets.xcassets/AppIcon.appiconset/256 1.png:1]: Why do we need both `256.png` and `256 1.png`?

**Review by tomekzaw — APPROVED** [review]

File: MacOSExample/macos/MacOSExample-macOS/Assets.xcassets/AppIcon.appiconset/256 1.png
Comment (jfedak) [reply MacOSExample/macos/MacOSExample-macOS/Assets.xcassets/AppIcon.appiconset/256 1.png:1]: I'm not sure why, but Xcode creates a duplicate for some images (same resolution but different scale). Maybe it's possible to change it in Xcode somehow, but I decided to leave it as it is.

Comment (jfedak) [reply MacOSExample/macos/MacOSExample-macOS/Assets.xcassets/AppIcon.appiconset/256 1.png:1]: Allright, I changed names from `256 1.png` to `256_1.png` to get rid of those spaces in file names.

Comment (tomekzaw) [reply MacOSExample/macos/MacOSExample-macOS/Assets.xcassets/AppIcon.appiconset/256 1.png:1]: Let's keep it the default way (i.e. what Xcode does by default).

**Pull request merged**